src/ResourceManager/Websites/Commands.Websites.Test/ScenarioTests/WebAppSlotTests.ps1

```diff
@@ -749,3 +749,81 @@ function Test-ManageSlotSlotConfigName
 	Assert-AreEqual 0 $slotConfigNames.ConnectionStringNames.Count
 }
+
+
+<#
+.SYNOPSIS
+Tests regular web app slot swap
+#>
+function Test-WebAppRegularSlotSwap
+{
+	$rgname = "Default-Web-EastAsia"
+	$appname = "webappslottest"
+	$sourceSlotName = "staging"
+	$destinationSlotName = "production"
+
+	# Swap Web App slots
+	$webApp = Swap-AzureRmWebAppSlot -ResourceGroupName $rgname -Name  $appname -SourceSlotName $sourceSlotName -DestinationSlotName $destinationSlotName
+}
+
+<#
+.SYNOPSIS
+Tests web app slot swap with preview: apply slot config followed by reset slot swap
+#>
+function Test-WebAppSwapWithPreviewResetSlotSwap
+{
+	Test-SlotSwapWithPreview 'ResetSlotSwap'
+}
+
+<#
+.SYNOPSIS
+Tests web app slot swap with preview: apply slot config followed by complete slot swap
+#>
+function Test-WebAppSwapWithPreviewCompleteSlotSwap
+{
+	Test-SlotSwapWithPreview 'CompleteSlotSwap'
+}
+
+<#
+.SYNOPSIS
+Test slot swap with preview feature
+#>
+function Test-SlotSwapWithPreview($swapWithPreviewAction)
+{
+    $rgname = "Default-Web-EastAsia"
+	$appname = "webappslottest"
+	$sourceSlotName = "staging"
+	$destinationSlotName = "production"
+	$appSettingName = 'testappsetting'
+	$originalSourceAppSettingValue = "staging"
+	$originalDestinationAppSettingValue = "production"
+
+	# Let's retrieve slot configs and make sure that it contains initial values as expected
+	$destinationWebApp = Get-AzureRmWebApp -ResourceGroupName $rgname -Name  $appname
+	Validate-SlotSwapAppSetting $destinationWebApp $appSettingName $originalDestinationAppSettingValue
+	
+	$sourceWebApp = Get-AzureRmWebAppSlot -ResourceGroupName $rgname -Name  $appname -Slot $sourceSlotName
+	Validate-SlotSwapAppSetting $sourceWebApp $appSettingName $originalSourceAppSettingValue
+
+	# Let's apply slot config and make sure that app setting values have been swapped
+	Swap-AzureRmWebAppSlot -ResourceGroupName $rgname -Name  $appname -SourceSlotName $sourceSlotName -DestinationSlotName $destinationSlotName -SwapWithPreviewAction 'ApplySlotConfig'
+	Start-Sleep -s 30
+	$sourceWebApp = Get-AzureRmWebAppSlot -ResourceGroupName $rgname -Name  $appname -Slot $sourceSlotName
+	Validate-SlotSwapAppSetting $sourceWebApp $appSettingName $originalDestinationAppSettingValue
+
+	# Let's finish the current slot swap operation (complete or reset)
+	Swap-AzureRmWebAppSlot -ResourceGroupName $rgname -Name  $appname -SourceSlotName $sourceSlotName -DestinationSlotName $destinationSlotName -SwapWithPreviewAction $swapWithPreviewAction
+	Start-Sleep -s 30
+	$sourceWebApp = Get-AzureRmWebAppSlot -ResourceGroupName $rgname -Name  $appname -Slot $sourceSlotName
+	Validate-SlotSwapAppSetting $sourceWebApp $appSettingName $originalSourceAppSettingValue
+}
+
+<#
+.SYNOPSIS
+Validates slot app setting for slot swap tests
+#>
+function Validate-SlotSwapAppSetting($webApp, $appSettingName, $expectedValue)
+{
+    Assert-AreEqual $webApp.SiteConfig.AppSettings[0].Name $appSettingName
```

Comment:
As the file is already using tabs instead of spaces, it's better to continue that coding style.